lib/run/summary.js

```diff
@@ -267,13 +267,25 @@ _.assign(RunSummary, {
     attachTimingTrackers (summary, emitter) {
         // mark the point when the run started
         // also mark the point when run completed and also store error if needed
-        emitter.on('start', function () { summary.run.timings.started = Date.now(); });
+        emitter.on('start', function () {
+            summary.run.timings.started = Date.now();
+            summary.run.timings.iterations = [];
+        });
         emitter.on('beforeDone', function () {
             summary.run.timings.completed = Date.now();
         });
         emitter.on('done', function (err) {
             err && (summary.error = err);
         });
+        emitter.on('beforeIteration', function (_err, o) {
+            summary.run.timings.iterations && summary.run.timings.iterations.push({
```

Comment:
Just need to quick check two things here:

1. Is this timings data structure acceptable now? @DannyDainton @codenirvana 
2. Why's there a need to `+ 1` on the iteration number? @adityaofficial10